ckEditor/src/ckeditor5-poll/poll-form.ts

```diff
@@ -0,0 +1,657 @@
+import Plugin from "@ckeditor/ckeditor5-core/src/plugin";
+import ClickObserver from "@ckeditor/ckeditor5-engine/src/view/observer/clickobserver";
+import ContextualBalloon from "@ckeditor/ckeditor5-ui/src/panel/balloon/contextualballoon";
+import clickOutsideHandler from "@ckeditor/ckeditor5-ui/src/bindings/clickoutsidehandler";
+
+import View from "@ckeditor/ckeditor5-ui/src/view";
+import ViewCollection from "@ckeditor/ckeditor5-ui/src/viewcollection";
+import InputTextView from "@ckeditor/ckeditor5-ui/src/inputtext/inputtextview";
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+
+import KeystrokeHandler from "@ckeditor/ckeditor5-utils/src/keystrokehandler";
+import FocusTracker from "@ckeditor/ckeditor5-utils/src/focustracker";
+import FocusCycler, { FocusableView } from "@ckeditor/ckeditor5-ui/src/focuscycler";
+
+import type { Editor } from "@ckeditor/ckeditor5-core";
+import type { Locale } from "@ckeditor/ckeditor5-utils";
+import type { Element, RootElement } from "@ckeditor/ckeditor5-engine";
+import type Writer from "@ckeditor/ckeditor5-engine/src/model/writer";
+
+import submitHandler from "@ckeditor/ckeditor5-ui/src/bindings/submithandler";
+
+import './poll.css';
+import { PollProps } from "./constants";
+
+export const POLL_COLOR_SCHEMES: PollProps['colorScheme'][] = [
+  { darkColor: '#06005C', lightColor: '#FFFFFF', bannerTextColor: '#FFFFFF'},
+  { darkColor: '#1D2A17', lightColor: '#FFFFFF', bannerTextColor: '#FFFFFF'},
+  { darkColor: '#7B3402', lightColor: '#FFFFFF', bannerTextColor: '#FFFFFF'},
+  { darkColor: '#F3F3E1', lightColor: '#222222', bannerTextColor: '#222222'},
+]
+
+class MainFormView extends View {
+  editor: Editor
+  selectedElement: (Element | RootElement)
+  questionView: InputTextView
+  agreeWordingView: InputTextView
+  disagreeWordingView: InputTextView
+  colorSchemeButtons: ButtonView[]
+  daysInputView: InputTextView
+  hoursInputView: InputTextView
+  minutesInputView: InputTextView
+  _focusables: ViewCollection<FocusableView>
+  focusTracker: FocusTracker
+  keystrokes: KeystrokeHandler
+  _focusCycler: FocusCycler
+  locale: Locale
+
+  constructor(locale: Locale, editor: Editor) {
```

Comment:
The UI doesn't look great in dark mode. Not sure if that's tricky to change or not?

<img width="655" alt="Screenshot 2025-04-25 at 12 58 35" src="https://github.com/user-attachments/assets/5f70c171-cb17-45f9-9798-a6df1758126c" />